src/ml_gym/frontend/dashboard/src/routes/analysis_board.tsx

```diff
@@ -0,0 +1,126 @@
+import "./analysis_board.css";
+
+import React from "react";
+import {
+    LineChart,
+    Line,
+    XAxis,
+    YAxis,
+    CartesianGrid,
+    Tooltip,
+    Legend
+} from "recharts";
+import type { RootState } from '../app/store';
+import { ModelEvaluationType, ModelEvaluationPayloadType, FilterConfigType } from "../app/datatypes"
+import { useAppSelector } from "../app/hooks"
+
+
+function selectColor(index: number) {
+    const hue = index * 137.508; // use golden angle approximation
+    return `hsl(${hue},50%,75%)`;
+}
+
+type AnalysisBoardProps = {
+    filterConfig: FilterConfigType
+};
+
+
+
+export const modelEvaluationSelector = (state: RootState) => state.modelsEvaluation.messages.reduce((results: any, s: ModelEvaluationType) => {
+    // results: {metric_key_1: [{name: 0, experiment_id_1: 0.1, ..., experiment_id_n: 0.4}, {name: 1, experiment_id_2: ...}, ...]}
+
+    const modelEvaluationPayload: ModelEvaluationPayloadType = s.data.payload
+
+    const epoch = modelEvaluationPayload.epoch
+    const eId = modelEvaluationPayload.experiment_id
+    const lossScores = modelEvaluationPayload.loss_scores
+    const metricScores = modelEvaluationPayload.metric_scores
+
+    for (const metricScore of metricScores) {
+        const fullMetricName = metricScore.split + "/" + metricScore.metric
+        if (!results[fullMetricName]) {
+            results[fullMetricName] = [{ name: epoch }]
+        }
+        results[fullMetricName][epoch] = { ...results[fullMetricName][epoch], [eId]: metricScore.score }
+    }
+
+    return results
+}, {});
+
+export const filteredModelEvaluationSelector = (state: RootState, metricFilter: string) => {
+    // results: {experiment_id_1: [0.9, 0.3, ...]}
+    const results = modelEvaluationSelector(state)
+    if (results[metricFilter]) {
+        return results[metricFilter]
+    } else {
+        return []
```

Comment:
Codacy has a fix for the issue: [Replace `········return·[]` with `····return·[];`](https://app.codacy.com/gh/mlgym/mlgym/pullRequest?prid=9966992)
```suggestion
    return [];
```